taskrun/ClusterTask.py

```diff
@@ -231,19 +233,19 @@ def execute(self):
     See Task.execute()
     """
 
-    # If we're killed at this point, don't bother starting a new process.
-    if self.killed:
-      return None
+    with self._lock:
+      # If we're killed at this point, don't bother starting a new process.
+      if self.killed:
+        return None
 
-    # start the command
-    cmd = self._build_command()
-    self._proc = subprocess.Popen(
-      cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+      # start the command
+      cmd = self._build_command()
+      self._proc = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True,
+        start_new_session=True)
 
-    # We could be killed while we're starting the process above. In this case,
-    # terminate the new process, but still clean up.
-    if self.killed:
-      self._proc.terminate()
+      # ensures that the subprocess is in a different group
+      assert os.getsid(self._proc.pid) != os.getsid(os.getpid())
```

Comment:
is this necessary? seems like a debugging-printf kind of thing
ditto in ProcessTask

Reply:
Removed.